### PR TITLE
Red lib fixes

### PR DESCRIPTION
**Pull request by Ivan-dev-RED**

### CODE REVIEW

File: src/redefined.resolver.ts [hunk 1]
```diff
@@ -38,21 +33,50 @@ export class RedefinedResolver implements Resolver {
...
 
     async resolve(domain: string, networks?: Network[]): Promise<Account[]> {
-        return flatten(await Promise.all(this.resolverServices.map(resolver =>
-          resolver.resolveAll(domain, this.nodes)
-        ))).filter(it => !networks || networks.includes(it.network));
+        const resolvers = flatten(this.resolverServices.map(name => this.resolvers[name]));
+        
+        const resolvedAccounts = flatten(await Promise.all(resolvers.map(it => it.resolve(domain))));
+        
+        const targetAccountsWithoutEvm = resolvedAccounts.filter(it => (!networks || networks.includes(it.network) && it.network !== "evm"));
```
Comment (Rednoll) [inline src/redefined.resolver.ts:59]: Default EVM logic can be moved to our resolvers from this global resolver

Comment (Rednoll) [reply src/redefined.resolver.ts:59]: It will be simpler and will no affect logic of ENS and Unst

**Review by Rednoll — CHANGES_REQUESTED** [review]

File: src/services/resolvers/redefined-resolver.service.ts [hunk 2]
```diff
@@ -1,32 +1,60 @@
...
                 from: "redefined"
             }));
         } catch (e: any) {
-            console.error("redefined Error", e.message);
-            return [];
+            throw Error(`redefined Error: ${e.message}`);
+        }
+    }
+
+    private async resolveEmail(domain: string): Promise<AccountRecord[]> {
```
[the remark this answers is not in the document]
Comment (Ivan-dev-RED) [reply src/services/resolvers/redefined-resolver.service.ts:35]: ok

File: src/services/resolvers/redefined-resolver.service.ts [hunk 3]
```diff
@@ -1,32 +1,60 @@
...
+            return await contract.methods.resolve(sha256(domain)).call();
+        } catch (e: any) {
+            if (e.message.includes("Name is not registered")) {
+                return [];
+            }
+            throw Error(`redefined Error: ${e.message}`);
+        }
+    }
+
+    private async resolveNickname(domain: string): Promise<AccountRecord[]> {
```
[the remark this answers is not in the document]
Comment (Ivan-dev-RED) [reply src/services/resolvers/redefined-resolver.service.ts:48]: ok

File: src/services/resolvers/unstoppable-resolver.service.ts [hunk 4]
```diff
@@ -5,29 +5,27 @@ import Resolution from "@unstoppabledomains/resolution";
 const resolution = new Resolution();
```
[the remark this answers is not in the document]
Comment (Ivan-dev-RED) [reply src/services/resolvers/unstoppable-resolver.service.ts:5]: ok

File: tests/moks/web3.ts [hunk 5]
```diff
@@ -10,18 +10,18 @@ EvmWeb3Service.getWeb3 = jest.fn().mockImplementation(
...
-                          ];
-                      },
-                  }
-              }
+        methods = {
+          resolve(domain: string) {
+            return {
+              async call(): Promise<AccountRecord[]> {
+                return [
+                    { addr: "0x123", network: "eth" },
```
[the remark this answers is not in the document]
Comment (Ivan-dev-RED) [reply tests/moks/web3.ts:18]: ok

**Review by Rednoll — APPROVED** [review]

**Review by Rednoll — DISMISSED** [review]

**Review by Rednoll — APPROVED** [review]

**Pull request merged**